### PR TITLE
docs: Organize examples as patches, add <Link> examples

### PR DESCRIPTION
**Pull request by toerndev**
- Code examples moved from examples/docs to examples.

- Each example may contain a README.md and a .patch file.

- Refer to examples as "recipes" in the main file.

- Remove example code from the main README.md, except for "minimal
  example". Instead use relative links to ./example/<recipe>.

- Added Link and NavLink examples.

### CODE REVIEW

**Comment by toerndev** [comment]
Oh, I need some configuration to disable eslint in ./examples. Ideally I'd like to still have auto-formatting based on .eslintrc to keep a consistent style.

File: README.md [hunk 1]
```diff
@@ -124,112 +124,24 @@ const NotFound = () => <h3>404</h3>
...
-`react-universal-component`, `babel-plugin-universal-import`, `webpack-flush-chunks`.
+- *Dispatching thunks on route changes*
+- *Performing redirects bases on `state` and `payload`.*
+- *Use hash-based routes/history (*see the [migration instructions](./docs/migration.md)*)*
+- *Restoring scroll position*
+- *Handling optional URL fragments and query strings*
+- *Route change pre- & post-processing*
+- *Code-splitting*
+- *Server-side rendering*
+- *Usage together with `react-universal-component`, `babel-plugin-universal-import`, `webpack-flush-chunks`.*
```
Comment (hedgepigdaniel) [inline README.md]: The last three are nicely demonstrated in `redux-first-router-link` - perhaps its worth adding a link to that?

Comment (toerndev) [reply README.md]: Sounds good, will check those out as the next step after this. Would be awesome to finish that list and then get started on ./docs!

Comment (toerndev) [reply README.md]: @hedgepigdaniel I added links to redux-first-router-link for the "links" examples. But I didn't find any examples of how to do the last three items above (SSR & universal etc), were you thinking of another repo?

**Pull request merged**